ComplianceTool NoSuchFileException when release name contains "/"

### Summary of the Bug
After changing creation of the clearing document name from using an ID to using the name and release version the test was executed with a release "testRelease". The test works. 
However, when a release has a more realistic name like "testRelease.com/release-commons", the test fails with:

```
org.eclipse.sw360.antenna.sw360.client.utils.SW360ClientException: Could not create clearing document /tmp/junit14748518631247097731/junit6381024443787852437/testRelease.com/release-commons1.0.0_clearing.json

	at org.eclipse.sw360.antenna.frontend.compliancetool.sw360.updater.ClearingReportGenerator.createClearingDocument(ClearingReportGenerator.java:32)
	at org.eclipse.sw360.antenna.frontend.compliancetool.sw360.updater.ClearingReportGeneratorTest.createClearingDocument(ClearingReportGeneratorTest.java:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.nio.file.NoSuchFileException: /tmp/junit14748518631247097731/junit6381024443787852437/testRelease.com/release-commons1.0.0_clearing.json
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:219)
	at java.base/java.nio.file.Files.newBufferedWriter(Files.java:2919)
	at java.base/java.nio.file.Files.newBufferedWriter(Files.java:2962)
	at org.eclipse.sw360.antenna.frontend.compliancetool.sw360.updater.ClearingReportGenerator.createClearingDocument(ClearingReportGenerator.java:29)
	... 28 more
```
Since the slash in the name is mistaken for a sub-directory, that does not exist. 

### Steps to reproduce
- Replace string ["testRelease" in `ClearingReportGeneratorTest` line 42](https://github.com/eclipse/antenna/blob/624cea5fdc21937b2efdb8bd3ab41571a6963ba5/assembly/compliance-tool/src/test/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/updater/ClearingReportGeneratorTest.java#L42) with "testRelease.com/release-commons" 

### Acceptance Criteria
- [ ] Clearing Documents are generated properly and uploaded
- [ ] Tests should be done with a live test sw360 instance, to assure it's working. 

### Definition of Done
- Acceptance criteria fulfilled
- A test case is created to reproduce the bug
- A PR is created, the CI infrastructure reports green, the bug test case proves that bug is fixed
- The PR is reviewed and approved
- No TODOs left in the code unless explained in the ticket, if something else is still open, this is summarized in a comment in the issue
- Documentation is updated

--- a/assembly/compliance-tool/src/main/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/updater/ClearingReportGenerator.java
+++ b/assembly/compliance-tool/src/main/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/updater/ClearingReportGenerator.java
@@ -22,7 +22,8 @@ public class ClearingReportGenerator {
     private static final String CLEARING_DOC_SUFFIX = "_clearing.json";
 
     Path createClearingDocument(SW360Release release, Path targetDir) {
-        Path clearingDocument = targetDir.resolve(release.getName() + release.getVersion() + CLEARING_DOC_SUFFIX);
+        String clearingDocumentName = release.getName() + "_" + release.getVersion() + CLEARING_DOC_SUFFIX;
+        Path clearingDocument = targetDir.resolve(clearingDocumentName.replaceAll("[/\\\\]", "_"));
         try {
             Files.createDirectories(targetDir);
             ObjectMapper mapper = new ObjectMapper();

--- a/assembly/compliance-tool/src/test/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/updater/ClearingReportGeneratorTest.java
+++ b/assembly/compliance-tool/src/test/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/updater/ClearingReportGeneratorTest.java
@@ -26,6 +26,9 @@ import java.nio.file.Path;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class ClearingReportGeneratorTest {
+    private static final String TEST_RELEASE_NAME = "testRelease.com/release\\commons";
+    private static final String EXPECTED_CLEARING_DOC_NAME = "testRelease.com_release_commons_1.0.0_clearing.json";
+
     @Rule
     public TemporaryFolder folder = new TemporaryFolder();
 
@@ -39,12 +42,12 @@ public class ClearingReportGeneratorTest {
     @Test
     public void createClearingDocument() throws IOException {
         File targetFolder = folder.newFolder();
-        SW360Release release = SW360TestUtils.mkSW360Release("testRelease");
+        SW360Release release = SW360TestUtils.mkSW360Release(TEST_RELEASE_NAME);
         release.setOverriddenLicense("overridden");
 
         Path clearingDocument = generator.createClearingDocument(release, targetFolder.toPath());
         assertThat(Files.exists(clearingDocument)).isTrue();
-        assertThat(clearingDocument.getFileName().toString()).isEqualTo(release.getName() + release.getVersion() + "_clearing.json");
+        assertThat(clearingDocument.getFileName().toString()).isEqualTo(EXPECTED_CLEARING_DOC_NAME);
 
         ObjectMapper mapper = new ObjectMapper();
         mapper.readValue(clearingDocument.toFile(), SW360Release.class);

